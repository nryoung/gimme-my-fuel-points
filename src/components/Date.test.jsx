import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import Date from './date';

it('renders date correctly', () => {
  const tree = shallow(
    <Date
      date={moment('2017-09-06')}
      onDateChange={() => {}}
      focused={true}
      onFocusChange={() => {}}
    />
  );
  expect(tree).toMatchSnapshot();
});
