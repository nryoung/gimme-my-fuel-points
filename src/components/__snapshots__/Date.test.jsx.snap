// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders date correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SingleDatePicker
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"2017-09-06T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={true}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    orientation="horizontal"
    phrases={
        Object {
            "chooseAvailableDate": [Function],
            "clearDate": "Clear Date",
            "closeDatePicker": "Close",
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month",
            "jumpToPrevMonth": "Move backward to switch to the previous month",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
          select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
            "moveFocusByOneMonth": "Switch months",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
            "openThisPanel": "Open this panel",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field",
            "selectFocusedDate": "Select the date in focus",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
            "upArrowDownArrow": "up and down arrow keys",
          }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={true}
    withFullScreenPortal={false}
    withPortal={false}
/>,
  "nodes": Array [
    <SingleDatePicker
      anchorDirection="left"
      customCloseIcon={null}
      customInputIcon={null}
      date={"2017-09-06T00:00:00.000Z"}
      daySize={39}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={true}
      horizontalMargin={0}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      orientation="horizontal"
      phrases={
            Object {
                  "chooseAvailableDate": [Function],
                  "clearDate": "Clear Date",
                  "closeDatePicker": "Close",
                  "dateIsUnavailable": [Function],
                  "enterKey": "Enter key",
                  "escape": "Escape key",
                  "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
                  "homeEnd": "Home and end keys",
                  "jumpToNextMonth": "Move forward to switch to the next month",
                  "jumpToPrevMonth": "Move backward to switch to the previous month",
                  "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
              select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                  "keyboardShortcuts": "Keyboard Shortcuts",
                  "leftArrowRightArrow": "Right and left arrow keys",
                  "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
                  "moveFocusByOneMonth": "Switch months",
                  "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
                  "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
                  "openThisPanel": "Open this panel",
                  "pageUpPageDown": "page up and page down keys",
                  "questionMark": "Question mark",
                  "returnFocusToInput": "Return to the date input field",
                  "selectFocusedDate": "Select the date in focus",
                  "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
                  "upArrowDownArrow": "up and down arrow keys",
                }
      }
      placeholder="Date"
      readOnly={false}
      renderCalendarInfo={null}
      renderDay={null}
      renderMonth={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={true}
      withFullScreenPortal={false}
      withPortal={false}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Date
        date={"2017-09-06T00:00:00.000Z"}
        onDateChange={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Date {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleFocusChange": [Function],
        "props": Object {
          "date": "2017-09-06T00:00:00.000Z",
          "onDateChange": [Function],
        },
        "refs": Object {},
        "state": Object {
          "focused": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"2017-09-06T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={true}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
                    Object {
                              "chooseAvailableDate": [Function],
                              "clearDate": "Clear Date",
                              "closeDatePicker": "Close",
                              "dateIsUnavailable": [Function],
                              "enterKey": "Enter key",
                              "escape": "Escape key",
                              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
                              "homeEnd": "Home and end keys",
                              "jumpToNextMonth": "Move forward to switch to the next month",
                              "jumpToPrevMonth": "Move backward to switch to the previous month",
                              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
                      select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                              "keyboardShortcuts": "Keyboard Shortcuts",
                              "leftArrowRightArrow": "Right and left arrow keys",
                              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
                              "moveFocusByOneMonth": "Switch months",
                              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
                              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
                              "openThisPanel": "Open this panel",
                              "pageUpPageDown": "page up and page down keys",
                              "questionMark": "Question mark",
                              "returnFocusToInput": "Return to the date input field",
                              "selectFocusedDate": "Select the date in focus",
                              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
                              "upArrowDownArrow": "up and down arrow keys",
                            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={true}
          withFullScreenPortal={false}
          withPortal={false}
/>,
        "_debugID": 2,
        "_renderedOutput": <SingleDatePicker
          anchorDirection="left"
          customCloseIcon={null}
          customInputIcon={null}
          date={"2017-09-06T00:00:00.000Z"}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          firstDayOfWeek={null}
          focused={false}
          hideKeyboardShortcutsPanel={true}
          horizontalMargin={0}
          id="date"
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepOpenOnDateSelect={false}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          numberOfMonths={1}
          onClose={[Function]}
          onDateChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          orientation="horizontal"
          phrases={
                    Object {
                              "chooseAvailableDate": [Function],
                              "clearDate": "Clear Date",
                              "closeDatePicker": "Close",
                              "dateIsUnavailable": [Function],
                              "enterKey": "Enter key",
                              "escape": "Escape key",
                              "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
                              "homeEnd": "Home and end keys",
                              "jumpToNextMonth": "Move forward to switch to the next month",
                              "jumpToPrevMonth": "Move backward to switch to the previous month",
                              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
                      select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                              "keyboardShortcuts": "Keyboard Shortcuts",
                              "leftArrowRightArrow": "Right and left arrow keys",
                              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
                              "moveFocusByOneMonth": "Switch months",
                              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
                              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
                              "openThisPanel": "Open this panel",
                              "pageUpPageDown": "page up and page down keys",
                              "questionMark": "Question mark",
                              "returnFocusToInput": "Return to the date input field",
                              "selectFocusedDate": "Select the date in focus",
                              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
                              "upArrowDownArrow": "up and down arrow keys",
                            }
          }
          placeholder="Date"
          readOnly={false}
          renderCalendarInfo={null}
          renderDay={null}
          renderMonth={null}
          reopenPickerOnClearDate={false}
          required={false}
          screenReaderInputMessage=""
          showClearDate={false}
          showDefaultInputIcon={true}
          withFullScreenPortal={false}
          withPortal={false}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Date
    date={"2017-09-06T00:00:00.000Z"}
    onDateChange={[Function]}
/>,
}
`;
